checks/pinned_dependencies.go

```diff
@@ -523,19 +522,14 @@ func validateGitHubWorkflowIsFreeOfInsecureDownloads(pathfn string, content []by
 
 			// We replace the `${{ github.variable }}` to avoid shell parsing failures.
 			script := githubVarRegex.ReplaceAll([]byte(run), []byte("GITHUB_REDACTED_VAR"))
-			scriptContent = fmt.Sprintf("%v\n%v", scriptContent, string(script))
-		}
-	}
-
-	if scriptContent != "" {
-		var err error
-		validated, err = validateShellFile(pathfn, []byte(scriptContent), dl)
-		if err != nil {
-			return false, err
+			validated, err := validateShellFile(pathfn, uint(execRun.Run.Pos.Line), 0 /*unknown*/, []byte(script), dl)
```

Comment:
I believe the reason the `TestGitHubWorflowRunDownload/wget_across_steps` test is failing is because we're now calling `validateShellFile` for each step instead of waiting to call it after we've gotten the `run` actions for all steps. A similar problem probably also would occur for Dockerfiles when something is downloaded and executed in different steps, although it doesn't look like we have a test to catch that.

Reply:
correct. I will fix it. I'll let you know once I have.

Reply:
Done, PTAL. I've simply added a `taintedFiles = map[string]bool` as input argument to keep track f downloaded files across invocations.